src/Raven.Server/Rachis/RachisLogHistory.cs

```diff
@@ -489,5 +498,141 @@ public unsafe bool HasHistoryLog(ClusterOperationContext context, BlittableJsonR
                 return true;
             }
         }
+
+        public bool TryGetLastDeleteDatabaseCommandWithoutErrors(ClusterOperationContext context, string nodeTag, string databaseName, out long index)
+        {
+            var commandType = nameof(DeleteDatabaseCommand);
+            index = 0;
+
+            var hasExceptions = false;
+
+            //DeleteDatabaseCommand id looks like that: DeleteDatabaseCommand/EncryptedTest_1/C/sadjghsdkjfng
+
+            var table = context.Transaction.InnerTransaction.OpenTable(LogHistoryTable, LogHistorySlice);
+
+            foreach (var result in table.SeekBackwardFrom(LogHistoryTable.Indexes[LogHistoryIndexSlice], Slices.AfterAllKeys))
+            {
+                var entryHolder = result.Result;
+                var curIndex = ReadIndex(entryHolder);
+                var guid = ReadGuid(entryHolder);
+                var type = ReadType(entryHolder);
+                var state = ReadState(entryHolder);
+
+                if (type == commandType && state == HistoryStatus.Committed &&
+                    TryGetNodesAndDatabaseFromDatabaseDeleteCommand(guid, out var databaseFromCmd, out var nodes) &&
+                    databaseFromCmd == databaseName &&
+                    (nodes.Count == 0 || nodes.Contains(nodeTag)))
+                {
+                    var exceptionType = ReadExceptionType(entryHolder);
+                    if (string.IsNullOrEmpty(exceptionType) == false)
+                    {
+                        if(index==0)
+                            hasExceptions = true;
+                        continue; // has errors after apply
+                    }
+
+                    hasExceptions = false;
+                    index = curIndex;
+                    break;
+                }
+            }
+
+
+            return hasExceptions == false;
+        }
+
+        private static bool TryGetNodesAndDatabaseFromDatabaseDeleteCommand(string guid, out string database, out List<string> nodes)
+        {
+            /*the DeleteDatabaseCommand command id should look like that: "db1/A_B_C/a6m8me9843jt"
```

Comment:
the format should be `DeleteDatabase/<database>/A_B_C/<guid>`

Reply:
Done